src/SignedSourceBase.hack

```diff
@@ -68,7 +72,10 @@ abstract class SignedSourceBase {
    *
    */
   public static function isSigned(string $file_data): bool {
-    return (bool)\preg_match(static::getPattern(), $file_data);
+    return C\any(
+      Regex\every_match($file_data, static::getPattern()),
+      $match ==> $match['token_name'] === static::getTokenName(),
+    );
```

Comment:
Is this a good way to avoid non-literal-string regexes?

It's a bit annoying.

Reply:
I think it's the best for now